lib/book.rb

```diff
@@ -0,0 +1,24 @@
+class Book
+  attr_reader :title, :author_first_name, :author_last_name, :publication_date
+
+  def initialize(book_info)
+    @title = book_info[:title]
+    @author_first_name = book_info[:author_first_name]
+    @author_last_name = book_info[:author_last_name]
+    @publication_date = book_info[:publication_date]
+  end
+
+  def author
+    @author_first_name + " " + @author_last_name
+  end
+
+  def publication_year
+    @publication_date[-4..-1]
+
+  end
+
```

Comment:
good job in here; you could simplify this by moving the string concatenation and the publication date work into your initialize method; we never asked that the author's first/last names be exposted, nor the full publication_date